diff --git a/resources/mmv/provider/mmv.provider.ImageInfo.js b/resources/mmv/provider/mmv.provider.ImageInfo.js
index 9973fe99..d47dccfb 100644
--- a/resources/mmv/provider/mmv.provider.ImageInfo.js
+++ b/resources/mmv/provider/mmv.provider.ImageInfo.js
@@ -89,17 +89,25 @@
 	 */
 	ImageInfo.prototype.get = function ( file ) {
 		var provider = this;
+		var params = {
+			action: 'query',
+			prop: 'imageinfo',
+			titles: file.getPrefixedDb(),
+			iiprop: provider.iiprop,
+			iiextmetadatafilter: provider.iiextmetadatafilter,
+			iiextmetadatalanguage: provider.options.language,
+			uselang: 'content'
+		};
 
-		return this.getCachedPromise( file.getPrefixedDb(), function () {
-			return provider.apiGetWithMaxAge( {
-				action: 'query',
-				prop: 'imageinfo',
-				titles: file.getPrefixedDb(),
-				iiprop: provider.iiprop,
-				iiextmetadatafilter: provider.iiextmetadatafilter,
-				iiextmetadatalanguage: provider.options.language,
-				uselang: 'content'
-			} ).then( function ( data ) {
+		if ( file.timestamp ) {
+			// To get the old image we have to reduce the timestamp and set the iistart for the imageinfo api.
+			params.iistart = file.timestamp -1;
+		}
+
+		return this.getCachedPromise( file.getPrefixedDb() + ( file.timestamp | '' ) , function () {
+			return provider.apiGetWithMaxAge(
+				params
+			).then( function ( data ) {
 				return provider.getQueryPage( file, data );
 			} ).then( function ( page ) {
 				if ( page.imageinfo && page.imageinfo.length ) {
