diff --git a/modules/ext.categoryTree/ext.categoryTree.js b/modules/ext.categoryTree/ext.categoryTree.js
index e4bac46..9b4288a 100644
--- a/modules/ext.categoryTree/ext.categoryTree.js
+++ b/modules/ext.categoryTree/ext.categoryTree.js
@@ -86,6 +86,11 @@
 	function attachHandler( $content ) {
 		$content.find( '.CategoryTreeToggle' )
 			.on( 'click', handleNode )
+			.on( 'keydown', function ( event ) {
+				if ( event.keyCode === 13 ) {
+					$( this ).click();
+				}
+			} )
 			.attr( 'title', function () {
 				return mw.msg(
 					$( this ).attr( 'data-ct-state' ) === 'collapsed' ?
